### ISSUE TITLE
Marks in the optional argument of `\twocolumn` are lost

### ISSUE THREAD

**Issue by pietvo** [latex3/latex2e#526]
## Brief outline of the bug

Marks in the optional argument of `\twocolumn`, for example when there is a `\section` command in it, are lost, and therefore are not refllected in the headers. In the example the header should be `Introduction`, but it is `2. First section` instead.

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{article}

  % Any preamble code goes here

\pagestyle{headings}
\renewcommand{\sectionmark}[1]{%
  \markright{\thesection. #1}%
}
\begin{document}

  % Demonstration of issue here
 
\twocolumn[\section{Introduction}\par Some text \par]

\section{First section}

Some text

\end{document}
```

## Log file (required) and possibly PDF file  

[testtwocolumnmarks.log](https://github.com/latex3/latex2e/files/6097831/testtwocolumnmarks.log)
[testtwocolumnmarks.pdf](https://github.com/latex3/latex2e/files/6097834/testtwocolumnmarks.pdf)

**Comment by muzimuzhi** [comment]
Optional argument of `\twocolumn` is typeset in a vbox, and since `\mark` commands can't escape boxes ...

Related:
 - TeX-SX question "[`\markboth` inside tikz `\node` text not working]"
 - TeX by Topic, Sec. 28.3 "Marks"
    > If the mark is given in horizontal mode it migrates to the surrounding vertical lists
like an insertion item (see page 76); however, if this is not the external vertical list,
the output routine will not find the mark.

**Comment by pietvo** [comment]
Yes, Frank, I fully understand that. But I just wanted to report it.
